acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/TekuNode.java

```diff
@@ -304,6 +338,19 @@ public void waitForNewFinalization() {
         MINUTES);
   }
 
+  public void waitForMilestone(SpecMilestone expectedMilestone) {
+    waitFor(
+        () -> {
+          final int currentEpoch = (int) getMetricValue("beacon_epoch");
+          final SpecMilestone currentMilestone =
+              spec.getForkSchedule().getSpecMilestoneAtEpoch(UInt64.valueOf(currentEpoch));
+          assertThat(currentMilestone).isEqualTo(expectedMilestone);
+          waitForNewFinalization();
```

Comment:
Why do we need to wait for the milestone to finalize?

It may be better to wait for the log line (`Activating network upgrade: CAPELLA`) that the milestone has been activated rather than just waiting for the epoch.

Reply:
Don't remember why I had it there, probably something I was using during my tests and did not remove.

I was using the log line before, the problem with the log line is that it does not work when a late node is syncing. So it only works for nodes that go through the upgrade.

Reply:
ah yeah for the other test that's an interesting case...  But getting the current epoch doesn't verify that it went through the transition at all, so there's probably no point in waiting for the transition to happen on a node that's not in sync. You'd just wait for it on the original node and know that they both have the same time so the second node must have the same current epoch anyway (then check they are actually in sync which we should already have a method for).

Reply:
So even if the node spec says that the upgrade is at epoch x, the node could be at epoch x + 1 without going through the upgrade? I would expect some kind of failure if it ever happens.

Reply:
You don't actually have the node spec from the running node though - you have the one from inside the DSL and while they should be the same, if there's a bug they may not be. So the only thing you're actually verifying is what the current epoch is and the AT already knows what epoch it scheduled the milestone to activate at and can just wait until the epoch without this method.